test/test_core.py

```diff
@@ -9418,15 +9418,8 @@ def test_Module_dynamicLibraries_pthreads(self):
     self.set_setting('NO_AUTOLOAD_DYLIBS')
 
     create_file('pre.js', '''
-      if (!global.Module) {
-        // This is the initial load (not a worker)
-        // Define the initial state of Module as we would
-        // in the html shell file.
-        // Use var to escape the scope of the if statement
-        var Module = {
-          dynamicLibraries: ['liblib.so']
-        };
-      }
+      // Load liblib.so by default
+      Module['dynamicLibraries'] = ['liblib.so'];
```

Comment:
I think the test is explicitly testing using extern-pre-js, with the idea that this will not get run on the workers.

Reply:
I see. I wasn't sure if this was done deliberately since the worker copies the `dynamicLibraries` from the main thread anyhow. See for example:
https://github.com/emscripten-core/emscripten/blob/057bc5534f59266328efed4c1e79640e15dc2bd1/src/worker.js#L157-L159

Changing the `--pre-js` confirms this:
```js
// Load liblib.so by default on non-workers
if (typeof importScripts == 'undefined') { // !ENVIRONMENT_IS_WORKER
  Module['dynamicLibraries'] = ['liblib.so'];
} else {
  console.log(Module['dynamicLibraries']); // outputs: [ 'liblib.so' ]
}
```

Reply:
How about extending this testcase to verify that? See e.g. commit 63610d1d589ae10575ff57218f71ffb997955619.

Reply:
I think that is exactly what this test case is verifying (or trying to verify) yes.